src/task/MonitorTask.ts

```diff
@@ -438,4 +462,42 @@ export default class MonitorTask
       poorConnectionCount: this.context.poorConnectionCount,
     };
   };
+
+  /**
+   * Degrade video codec to an alternative codec
+   */
+  private degradeVideoCodec(cause: string): void {
+    // Degrade video codec if there are other codec options and current codec is not H264 CBP or VP8
+    if (
+      this.context.meetingSupportedVideoSendCodecPreferences !== undefined &&
+      this.context.meetingSupportedVideoSendCodecPreferences.length > 1 &&
+      !(
+        this.context.meetingSupportedVideoSendCodecPreferences[0].equals(
+          VideoCodecCapability.h264ConstrainedBaselineProfile()
+        ) ||
+        this.context.meetingSupportedVideoSendCodecPreferences[0].equals(VideoCodecCapability.vp8())
+      )
+    ) {
+      const newMeetingSupportedVideoSendCodecPreferences: VideoCodecCapability[] = [];
+      for (const capability of this.context.videoSendCodecPreferences) {
+        if (!capability.equals(this.context.meetingSupportedVideoSendCodecPreferences[0])) {
+          newMeetingSupportedVideoSendCodecPreferences.push(capability);
+        }
+      }
+      if (newMeetingSupportedVideoSendCodecPreferences.length > 0) {
+        this.context.logger.info(
+          `Downgrading codec from ${this.context.meetingSupportedVideoSendCodecPreferences[0].codecName} to ${newMeetingSupportedVideoSendCodecPreferences[0].codecName} due to ${cause}`
+        );
+        this.context.degradedVideoSendCodecs.push(
+          this.context.meetingSupportedVideoSendCodecPreferences[0]
+        );
+        this.context.meetingSupportedVideoSendCodecPreferences = newMeetingSupportedVideoSendCodecPreferences;
+        this.context.audioVideoController.update({ needsRenegotiation: true });
+      } else {
+        this.context.logger.warn(
+          'Degrading video codec failed since there is no alternative codec to select'
```

Comment:
Could we dump all the currently degraded video codecs here?

Reply:
Changed log line to dump degraded codec here